gbpservice/nfp/orchestrator/config_drivers/heat_driver.py

```diff
@@ -445,6 +484,69 @@ def _get_all_heat_resource_keys(self, template_resource_dict,
                 resource_keys.append(key)
         return resource_keys
 
+    def _create_firewall_template(self, auth_token, service_details, stack_template):
+        provider = service_details['provider_ptg']
+
+        consuming_ptgs_details = service_details['consuming_ptgs_details']
+        consumer_eps = service_details['consuming_external_policies']
+
+        if (not consuming_ptg_details) and (not consumer_eps):
+            return None
+
+        is_template_aws_version = stack_template.get(
+            'AWSTemplateFormatVersion', False)
+        resources_key = 'Resources' if is_template_aws_version else 'resources'
+        properties_key = ('Properties' if is_template_aws_version
+                          else 'properties')
+        fw_rule_keys = self._get_all_heat_resource_keys(
+            stack_template[resources_key], is_template_aws_version,
+            'OS::Neutron::FirewallRule')
+        fw_policy_key = self._get_all_heat_resource_keys(
+            stack_template['resources'], is_template_aws_version,
+            'OS::Neutron::FirewallPolicy')[0]
+
+        provider_cidr = service_details['provider_subnet']['cidr']
+
+        fw_template_properties = dict(
+            resources_key=resources_key, properties_key=properties_key,
+            is_template_aws_version=is_template_aws_version,
+            fw_rule_keys=fw_rule_keys,
+            fw_policy_key=fw_policy_key)
+
+        for consumer in consuming_ptgs_details:
+            ptg = consumer['ptg']
+            subnets = consumer['subnets']
+
+            # Skip the stitching PTG
+            if ptg['proxied_group_id']:
+                continue
+
+            fw_template_properties.update({'name': ptg['id'][:3]})
+            for subnet in subnets:
+                if subnet['name'].startswith(APIC_OWNED_RES):
+                    continue
+
```

Comment:
provider_subnet can be None, Following check is missing
https://github.com/oneconvergence/group-based-policy/blob/mitaka_21st_march_base/gbpservice/nfp/orchestrator/config_drivers/heat_driver.py#L472